src/NUnitCore/core/Builders/ProviderReference.cs

```diff
@@ -12,56 +12,52 @@ namespace NUnit.Core.Builders
 {
     class ProviderReference
     {
-        private Type providerType;
-        private string providerName;
-        private object[] providerArgs;
-        private string category;
-
         public ProviderReference(Type providerType, string providerName, string category)
         {
             if (providerType == null)
                 throw new ArgumentNullException("providerType");
             if (providerName == null && providerType.GetInterface("System.Collections.IEnumerable") == null)
                 throw new ArgumentNullException("providerName");
 
-            this.providerType = providerType;
-            this.providerName = providerName;
-            this.category = category;
+            ProviderType = providerType;
+            ProviderName = providerName;
+            ProviderLocation = providerType.FullName + "." + providerName;
+            ProviderCategory = category;
         }
 
         public ProviderReference(Type providerType, object[] args, string providerName, string category)
             : this(providerType, providerName, category)
         {
-            this.providerArgs = args;
+            ProviderArgs = args;
         }
 
-        public string Name
-        {
-            get { return this.providerName; }
-        }
+        public Type ProviderType { get; private set; }
 
-        public string Category
-        {
-            get { return this.category; }
-        }
+        public string ProviderName { get; private set; }
+
+        public object[] ProviderArgs { get; private set; }
+
+        public string ProviderLocation { get; private set; }
+
+        public string ProviderCategory { get; private set; }
```

Comment:
Option, `{ get; }` has stronger guarantees (backed by readonly fields).

Reply:
@rprouse Exactly. I actually had the straight `{ get; }` originally, as well as a bunch of use of `?.` in various places. It worked in the IDE, since I'm using VS2015, but the nant build failed and I decided to revert rather than getting involved in a nant upgrade.